db/migrate/20200728054557_refer_config_proxy_to_http_proxy.rb

```diff
@@ -0,0 +1,75 @@
+class ReferConfigProxyToHttpProxy < ActiveRecord::Migration[6.0]
+  def up
+    add_column :foreman_virt_who_configure_configs, :http_proxy_id, :integer
+
+    if User.where(login: User::ANONYMOUS_ADMIN).exists?
+      User.as_anonymous_admin do
+        http_proxies = HttpProxy.all
+        ::ForemanVirtWhoConfigure::Config.find_each do |config|
+          if config.proxy.present?
+            ReferConfigProxyToHttpProxy.migrate_config(http_proxies, config)
+          end
+        end
+      end
+    end
+
+    remove_column :foreman_virt_who_configure_configs, :proxy
+  end
+
+  def down
+    add_column :foreman_virt_who_configure_configs, :proxy, :string
+
+    if User.where(login: User::ANONYMOUS_ADMIN).exists?
+      User.as_anonymous_admin do
+        ::ForemanVirtWhoConfigure::Config.find_each do |config|
+          if config.http_proxy_id.present?
+            config.proxy = config.http_proxy.url
+            config.save!
+          end
+        end
+      end
+    end
+
+    remove_column :foreman_virt_who_configure_configs, :http_proxy_id
+  end
+
+  class << self
```

Comment:
Since you are actually testing the full up and down methods, making these class methods doesn't apply per my previous suggestion. Feel free to make them instance methods again, but it's not a big deal IMO